public/docs/ts/latest/guide/dependency-injection.jade

```diff
@@ -592,55 +595,56 @@ code-example(format).
     The injector resolves these tokens and injects the corresponding services into the matching factory function parameters.
 :marked
   Notice that we captured the factory provider in an exported variable, `heroServiceProvider`.
-  This extra step makes the factory provider re-usable. 
-  We can register our `HeroService` with this variable whereever we need it.
-  
-  In our sample, we need it only in the `HeroesComponent` 
+  This extra step makes the factory provider reusable.
+  We can register our `HeroService` with this variable wherever we need it.
+
+  In our sample, we need it only in the `HeroesComponent`,
   where it replaces the previous `HeroService` registration in the metadata `providers` array.
   Here we see the new and the old implementation side-by-side:
 +makeTabs(
-  `dependency-injection/ts/app/heroes/heroes.component.ts, 
+  `dependency-injection/ts/app/heroes/heroes.component.ts,
   dependency-injection/ts/app/heroes/heroes.component.1.ts`,
-  null, 
+  null,
   `app/heroes/heroes.component (v.3),
   app/heroes/heroes.component (v.2)`)
-  
+
 <a id="token"></a>
 .l-main-section
 :marked
-  ## Dependency Injection Tokens
+  ## Dependency injection tokens
 
-  When we register a provider with an injector we associate that provider with a dependency injection token.
-  The injector maintains an internal *token/provider* map that it references when
+  When we register a provider with an injector, we associate that provider with a dependency injection token.
+  The injector maintains an internal *token-provider* map that it references when
   asked for a dependency. The token is the key to the map.
-  
-  In all previous examples, the dependency value has been a class *instance* and 
+
+  In all previous examples, the dependency value has been a class *instance*, and
   the class *type* served as its own lookup key.
-  Here we get a `HeroService` directly from the injector by supplying the `HeroService` type as the key/token.
+  Here we get a `HeroService` directly from the injector by supplying the `HeroService` type as the token:
 +makeExample('dependency-injection/ts/app/injector.component.ts','get-hero-service')(format='.')
 :marked
-  We have similar good fortune (in typescript) when we write a constructor that requires an injected class-based dependency.
-  We define a constructor parameter with the `HeroService` class type and Angular knows to inject the
+  We have similar good fortune when we write a constructor that requires an injected class-based dependency.
```

Comment:
I suppose this is true in both TS and Dart. In JS we'd have to annotate to tell Angular the service/provider lookup token.